packages/quantic/force-app/main/default/lwc/quanticNumericFacet/e2e/quanticNumericFacet.e2e.ts

```diff
@@ -0,0 +1,151 @@
+        baseFacet,
+        facet,
+      }) => {
+        const selectedIndex = 0;
+        const {facetId, field, values} = facetData;
+        const expectedFacetData = {
+          facetId,
+          facetField: field,
+          facetValue: `${values[selectedIndex].start}..${values[selectedIndex].end}`,
+        };
+
+        const facetSelectUaRequest =
+          baseFacet.waitForFacetSelectUaAnalytics(expectedFacetData);
+        let searchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.clickOnFacetValue(selectedIndex);
+
+        await facetSelectUaRequest;
+        let searchResponse = await searchResponsePromise;
+        const {analytics: analyticsForFacetSelect} =
+          baseFacet.extractDataFromResponse(searchResponse);
+        expect(analyticsForFacetSelect.actionCause).toEqual('facetSelect');
+        expect(analyticsForFacetSelect.customData).toEqual(
+          expect.objectContaining(expectedFacetData)
+        );
+
+        const clearAllUaRequest = baseFacet.waitForFacetClearAllUaAnalytics({
+          facetId,
+          facetField: field,
+        });
+        searchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.clickOnClearSelectionButton();
+
+        await clearAllUaRequest;
+        searchResponse = await searchResponsePromise;
+        const {analytics: analyticsForFacetDeselect} =
+          baseFacet.extractDataFromResponse(searchResponse);
+        expect(analyticsForFacetDeselect.actionCause).toEqual('facetClearAll');
+      });
+    });
+
+    test.describe('when typing in the facet search box input', () => {
+      test('should fetch facet values according to the query typed', async ({
+        baseFacet,
+        facet,
+      }) => {
+        const exampleMin = '1';
+        const exampleMax = '2';
+        const {facetId, field} = facetData;
+        const expectedFacetData = {
+          facetId: `${facetId}_input`,
+          facetField: field,
+          facetValue: `${exampleMin}..${exampleMax}`,
+        };
+
+        const facetSelectUaRequest =
+          baseFacet.waitForFacetSelectUaAnalytics(expectedFacetData);
+        let searchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.fillFilterMinInput(exampleMin);
+        await facet.fillFilterMaxInput(exampleMax);
+        await facet.clickOnFilterApplyButton();
+
+        await facetSelectUaRequest;
+        let searchResponse = await searchResponsePromise;
+        const {analytics: analyticsForFacetSelect} =
+          baseFacet.extractDataFromResponse(searchResponse);
+        expect(analyticsForFacetSelect.actionCause).toEqual('facetSelect');
+        expect(analyticsForFacetSelect.customData).toEqual(
+          expect.objectContaining(expectedFacetData)
+        );
+      });
+    });
+
+    if (useCase.value === useCaseEnum.search) {
+      test.describe('with a selected value in the URL', () => {
```

Comment:
do you think there is some value here in adding a test that checks the url is updated when we deselect a (previously selected) facet?

Reply:
It could be yes but I'm not sure as this may seem a bit as over testing so I wouldn't necessarily add such a test.